can/interfaces/vector/canlib.py

```diff
@@ -154,31 +188,64 @@ def set_filters(self, can_filters=None):
 
     def recv(self, timeout=None):
         end_time = time.time() + timeout if timeout is not None else None
-        event = vxlapi.XLevent(0)
-        event_count = ctypes.c_uint()
+
+        if self.fd:
+            event = vxlapi.XLcanRxEvent()
+        else:
+            event = vxlapi.XLevent()
+            event_count = ctypes.c_uint()
+            
         while True:
-            event_count.value = 1
-            try:
-                vxlapi.xlReceive(self.port_handle, event_count, event)
-            except VectorError as exc:
-                if exc.error_code != vxlapi.XL_ERR_QUEUE_IS_EMPTY:
-                    raise
+            if self.fd:
+                try:
+                    vxlapi.xlCanReceive(self.port_handle, event)
+                except VectorError as exc:
+                    if exc.error_code != vxlapi.XL_ERR_QUEUE_IS_EMPTY:
+                        raise
+                else:
+                    if event.tag == vxlapi.XL_CAN_EV_TAG_RX_OK or event.tag == vxlapi.XL_CAN_EV_TAG_TX_OK:
+                        msg_id = event.tagData.canRxOkMsg.canId
+                        dlc = self.map_dlc(event.tagData.canRxOkMsg.dlc)
```

Comment:
You can use `can.util.dlc2len()`.

Reply:
I will remove the own mapping and use the utils function.